test/integration/src/test/java/io/pravega/test/integration/endtoendtest/EndToEndChannelLeakTest.java

```diff
@@ -170,15 +170,16 @@ public void testDetectChannelLeakSegmentSealedPooled() throws Exception {
         event = reader1.readNextEvent(10000);
         assertNotNull(event.getEvent());
 
-        assertChannelCount(5, connectionPool);
+        //the append paths increases the connections
+        assertChannelCount(12, connectionPool);
```

Comment:
We are breaching the max channel count configured
`ClientConfig.builder().maxConnectionsPerSegmentStore(5)` which is supposed to be 5.

Reply:
yes.. this is because append paths will have dedicated connection.
I will raise another PR in which except the EventWriter rest other append paths will use connection pool.

Reply:
thanks for the clarification.

Reply:
We could update the javadoc of `maxConnectionsPerSegmentStore` to indicate that Connection pooling is disabled for writers.

Reply:
I will create another PR for this.